baseapp/baseapp.go

```diff
@@ -13,6 +13,7 @@ import (
 	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 	dbm "github.com/tendermint/tm-db"
 
+	txvalidate "github.com/cosmos/cosmos-sdk/client/tx"
```

Comment:
baseapp should not import anything from client, it creates messy code. Is there somewhere else we can put ValidateMsg?

Reply:
what about `types/utils.go` ?

Reply:
I believe we're moving away of putting anything in the `types` root package, it's too large already.

We can also create a new `types/msg` package. @robert-zaremba @aaronc Do you have any opinions on this?

Reply:
> or put it in types/bech32, and call it bech32.ValidateMsgSigners
I don't think it is fine to put it in `types/bech32`, since it is calling `msg.ValidateBasic` calling inside. For now I kept it in the `types/tx`.

Reply:
I would like to avoid adding these as new public functions if possible and that means we don't need to think about packages as much if they're private. Since we are deprecating global bech32's soon that will avoid another API breaking change.